jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -184,9 +185,9 @@ String getMessage()
     public static HttpURI createHttpURI(String scheme, String host, int port, String path, String param, String query, String fragment)
```

Comment:
So what does it mean if port==0 is passed in?  Should that be corrected to NO_PORT? or left as 0?

Reply:
port 0 is how we used to convey no port. (pre HttpURI object).
I'm tempted to reject port 0, but then again, there's cases where we create an HttpURI from something that isn't http. or on a network.
Either way, we shouldn't carry the port 0 thru to `java.net.URI` at all.